src/backend/joanie/signature/backends/lex_persona.py

```diff
@@ -469,25 +470,30 @@ def handle_notification(self, request: Request):
 
         reference_id = trusted_event_signature_provider.get("workflowId")
         event_type = trusted_event_signature_provider.get("eventType")
-        if event_type == "workflowFinished":
-            self.confirm_organization_signature(reference_id)
-        elif event_type == "recipientFinished":
-            self.confirm_student_signature(reference_id)
-        elif event_type == "recipientRefused":
-            self.reset_contract(reference_id)
-        else:
-            logger.error(
-                "'%s' is not an event type that we handle.",
-                event_type,
-                extra={
-                    "context": {
-                        "trusted_event_signature_provider": trusted_event_signature_provider
-                    }
-                },
-            )
-            raise ValidationError(
-                f"The notification {event_type} is not supported.",
-            )
+        try:
+            # ruff : noqa : BLE001
+            # pylint: disable=broad-exception-caught
+            if event_type == "workflowFinished":
+                self.confirm_organization_signature(reference_id)
+            elif event_type == "recipientFinished":
+                self.confirm_student_signature(reference_id)
+            elif event_type == "recipientRefused":
+                self.reset_contract(reference_id)
+            else:
+                logger.error(
+                    "'%s' is not an event type that we handle.",
+                    event_type,
+                    extra={
+                        "context": {
+                            "trusted_event_signature_provider": trusted_event_signature_provider
+                        }
+                    },
+                )
+                raise ValidationError(
+                    f"The notification {event_type} is not supported.",
+                )
+        except Exception as error:
```

Comment:
So far, since we don't yet really understand the specific exceptions we can encounter, I would say I am okay by catching Exception. This will bring us more understanding on what is occuring when an exception is catched.
But I feel like we should try to narrow it down to specific exceptions. Maybe it could be done later once we have more examples on our sentry instance. WDYT ?

Reply:
IMO it's fine here to catch all exceptions. If an error is raised inside this try...catch that means something is wrong in our own logic so it should not impact the response we return to Lex Persona.

Just before this `try...catch` statement, we are checking the received web hook event and in this case we are raising error if something is wrong. That makes sense as there is something wrong with the data coming from Lex Persona in this case.

Reply:
Perfect then ! :ok_hand: I actually find it accurate because we will catch every sort of error from our own logic.